domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/Frequency.java

```diff
@@ -0,0 +1,228 @@
+            EntityBuildResult<?> toReturn;
+            if (tripId == null || startTime == null || endTime == null || headwaySecs == null ||
+                    headwaySecs < 0 || exactTimes == null) {
+
+                if (tripId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "trip_id", tripId));
+                }
+
+                if (startTime == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "start_time", tripId));
+                }
+
+                if (endTime == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "end_time", tripId));
+                }
+
+                if (headwaySecs == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "headway_secs", tripId));
+                } else if (headwaySecs < 0) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("frequencies.txt",
+                            "headway_secs", tripId, 0, Integer.MAX_VALUE, headwaySecs));
+                }
+
+                // the following statement is true when ExactTimes.isEnumValueValid(originalExactTimes)
+                // returns false
+                if (exactTimes == null) {
+                    noticeCollection.add(new UnexpectedEnumValueNotice("frequencies.txt",
+                            "exact_times", tripId, originalExactTimes));
+                }
+                toReturn = new EntityBuildResult<>(noticeCollection);
+                noticeCollection.clear();
+            } else {
+                toReturn = new EntityBuildResult<>(new Frequency(tripId, startTime, endTime, headwaySecs, exactTimes));
+            }
+            return toReturn;
+        }
+    }
+
+    //Temporary implementation before TimeConversionUtils class implementation
+    public static Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsString) {
+        if (timeAsString != null && timeAsString.matches("([0-9][0-9]|[0-9]):[0-5][0-9]:[0-5][0-9]")) {
+            final String[] timeStringSplit = timeAsString.split(":");
+            try {
+                final int hourValue = Integer.parseInt(timeStringSplit[0]);
+                final int minuteValue = Integer.parseInt(timeStringSplit[1]);
+                final int secondValue = Integer.parseInt(timeStringSplit[2]);
+                // Converting time to an Integer value representing number of seconds
+                final int timeValueAsInt = hourValue * 3600 + minuteValue * 60 + secondValue;
+                // Setting noon as point zero, subtracting 12 * 3600 seconds
+                return timeValueAsInt - (12 * 3600);
+            } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+                return null;
```

Comment:
See my above comment - it would be good to have more comments about the interactions of this with type validation - for example, if we expect this to never happen because the string has already been validated as "HH:MM:SS" earlier in the validation pipeline.

Reply:
That code was implemented elsewhere, see 7577738eb1611d455514121fd20aebb3f5b31718